### ISSUE TITLE
Can Flet act as its own server to receive and handle http requests from an outside server?

### ISSUE THREAD

**Issue by JorgeAlvarez21** [flet-dev/flet#879]
Hi I'm building a chat app using Flask for the Server API. I am wondering if Flet is capable to handle **incoming** http requests from this outside server or maybe events directly from a database, so I can notify the user (if is online) and refresh the Flet components in real time.
I can successfully send them and receive the responses from the main app, but how can I make it available to receive them at random times? 

If Flet is not able to do this as of yet, my second approach would be to create a new thread in the main app file that sends constant requests to the API server and refreshes on any new updates. I tried to implement this second option but i encountered this error.
- Implementation
```

def run_flet_one_thread():
	# this runs the app
    ft.app(target=main, assets_dir="../../media", host='localhost', port=9000, view=ft.WEB_BROWSER)


def events_handler():
	#Testing thread
    running = True
    counter = 0
    while running:
        print('Running App')
        time.sleep(3)
        if counter >= 10:
            running = False
        counter += 1

if __name__ =="__main__":
    # creating thread
    thread_one = threading.Thread(target=run_flet_one_thread, name='flet_app')
    thread_two = threading.Thread(target=events_handler, name='events_handler')
    thread_one.start()
    thread_two.start()

```

 - Error being raised (weirdly enough both threads still seem to be running, the value error does not stop execution)
```
Running App
http://localhost:49573
Exception in thread flet_app:
Traceback (most recent call last):
  File "/opt/anaconda3/envs/chat-app-env/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/opt/anaconda3/envs/chat-app-env/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/joglnx/Documents/Projects/chat_app/root/frontend/main.py", line 165, in run_flet_one_thread
    ft.app(target=main, assets_dir="../../media", host='localhost', port=9000, view=ft.WEB_BROWSER,
  File "/opt/anaconda3/envs/chat-app-env/lib/python3.10/site-packages/flet/flet.py", line 119, in app
    signal.signal(signal.SIGINT, exit_gracefully)
  File "/opt/anaconda3/envs/chat-app-env/lib/python3.10/signal.py", line 56, in signal
    handler = _signal.signal(_enum_to_int(signalnum), _enum_to_int(handler))
ValueError: signal only works in main thread of the main interpreter
Running App

```

Any ideas on how to go about this?

**Comment by ndonkoHenri** [comment]
> how can I make it available to receive them at random times?

Async/Await maybe?

**Comment by FeodorFitsner** [comment]
Exposing API from a Flet is an interesting challenge - I haven't thought about that yet.

As I understand the main concern is that Flet uses its own web server, so having two web servers in the app is kind of weird. What if we could host Flet in an existing web server such as Flask or FastAPI?

**Comment by JorgeAlvarez21** [comment]
That's a good idea I'll give it a try in future projects since I'm not sure how I would go about doing that and most the logic is already done. 
I'm looking for a way to do a server push into Flet so I can update its controls accordingly. I think it sounds a bit problematic, so instead I want to try to implement ongoing server pulling from inside the Flet's main app to the Api, in a separate thread so they can run in parallel.

So the question now is do you have an example for properly executing a new thread inside the main app?

**Comment by ndonkoHenri** [comment]
It's pretty straightforward.
There's some code at the bottom here:
https://flet.dev/docs/guides/python/user-controls/

**Comment by JorgeAlvarez21** [comment]
Thanks for that link it helped!
All I needed was to pass daemon to prevent blocking of the main app thread.
